### ISSUE TITLE
NestedTryDepth false-positive

### ISSUE THREAD

**Issue by boris-petrov** [checkstyle/checkstyle#6201]
$ cat TestClass.java
```java
public class TestClass {
    void method() {
        try (var output = new PipedOutputStream()) {
            qwe(() -> {
                try (var input = new PipedInputStream(output)) {
                    try {
                        bar(input);
                    } finally {
                        baz(input);
                    }
                }
            });
        }
    }

    void bar(PipedInputStream input) {
    }

    void baz(PipedInputStream input) {
    }
}
```

$ cat TestConfig.xml
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
       <module name="NestedTryDepth" />
    </module>
</module>
```

$ java -jar checkstyle-8.14-all.jar -c TestConfig.xml TestClass.java
```
Starting audit...
[ERROR] /home/boris/TestClass.java:6:41: Nested try depth is 2 (max allowed is 1). [NestedTryDepth]
Audit done.
Checkstyle ends with 1 errors.
```

I think this should be considered as a bug - there is no way to rewrite this without a nested `try`. Note that if you remove the outermost `try` and the `qwe` method call - then the bug disappears.

**Comment by romani** [comment]
Usually such violations are fixed but moving nested TRYs to special method, is it really not possible in your case ?

**Comment by boris-petrov** [comment]
It is, I didn't know that this is what I should do. It is still a bug though, right?

**Comment by romani** [comment]
If you can move, it is not a bug. It might be not very good refactoring for readability but in this case it is "won't fix" and it is ok , you need to use suppression. Checkstyle is only bring your attention to a case, it is your call to fix it or suppress by a reason.

Style check is not bug search, some cases are fine, and better to stay as is.

**Comment by rnveach** [comment]
Since this doesn't look like a bug, I will close the issue.

Violations like this usually mean remove a try, move the try somewhere else (like different placement or new method), suppress it, or question the settings for the check (maybe you want to allow a nested depth up to 2 or remove the check completely).
It all depends on how you want your code to look, hence the naming 'style' in the name. :)
